rdr_service/dao/workbench_dao.py

```diff
@@ -850,6 +850,48 @@ def _validate(self, resource_json):
                         raise BadRequest(
                             f"Invalid nonAcademicAffiliation: {institution.get('nonAcademicAffiliation')}")
 
+    def get_redcap_audit_researchers(self, **filters):
+        last_snapshot_id = filters.get('last_snapshot_id', None)
+        snapshot_id = filters.get('snapshot_id', None)
+        user_source_id = filters.get('user_source_id', None)
+
+        results = []
+        with self.session() as session:
+            researchers = (
+                session.query(WorkbenchResearcher)
+            )
+            if user_source_id:
+                researchers = researchers.filter(WorkbenchResearcher.userSourceId == user_source_id) \
+                    .order_by(desc(WorkbenchResearcher.id)).limit(1)
+            elif snapshot_id:
+                researchers = researchers.filter(WorkbenchResearcher.id == snapshot_id)
+            elif last_snapshot_id:
+                researchers = researchers.filter(WorkbenchResearcher.id > last_snapshot_id) \
+                    .order_by(WorkbenchResearcher.id)
+
+            for researcher in researchers.all():
+                affiliations = []
+                if researcher.workbenchInstitutionalAffiliations:
+                    for affiliation in researcher.workbenchInstitutionalAffiliations:
+                        affiliations.append(
+                            {
+                                "institution": affiliation.institution,
+                                "role": affiliation.role,
+                                "isVerified": affiliation.isVerified,
+                                "nonAcademicAffiliation":
+                                    str(WorkbenchInstitutionNonAcademic(affiliation.nonAcademicAffiliation))
+                                    if affiliation.nonAcademicAffiliation else 'UNSET'
```

Comment:
```suggestion
                                "nonAcademicAffiliation":
                                    str(WorkbenchInstitutionNonAcademic(affiliation.nonAcademicAffiliation or 0))
```

nit: Maybe this could use an `or 0` to get UNSET if `nonAcademicAffiliation` is None. But I'm not sure which would be easier to read through.